library/common/http/client.cc

```diff
@@ -517,14 +517,16 @@ void Client::cancelStream(envoy_stream_t stream) {
     ScopeTrackerScopeState scope(direct_stream.get(), scopeTracker());
     removeStream(direct_stream->stream_handle_);
 
-    direct_stream->callbacks_->onCancel();
-
     // Since https://github.com/envoyproxy/envoy/pull/13052, the connection manager expects that
     // response code details are set on all possible paths for streams.
     direct_stream->setResponseDetails(getCancelDetails());
 
     // Only run the reset callback if the stream is still open.
     if (stream_was_open) {
+      ENVOY_LOG(debug, "[S{}] application cancelled stream", stream);
+
+      direct_stream->callbacks_->onCancel();
```

Comment:
This distinction with the callback only actually matters when the stream is being manually controlled; in the default case if the stream not being present in the map, it's equivalent to it not being open.